src/modules/poolModule.ts

```diff
@@ -653,7 +653,7 @@ export class PoolModule implements BaseModule {
     headers?: HeadersInit,
     validate: boolean = true,
   ): Promise<ExtendedPoolWithApr[]> {
-    const pools = await fetchAllPoolsApi(this.sdk.baseUrl, headers);
+    const pools = await fetchAllPoolsApi(this.sdk.baseUrl, headers ?? this.sdk.customHeaders);
```

Comment:
Instead of replacing the headers, suggest append the custom headers, and allow headers to override. 
```
{
  ...this.sdk.customHeaders, 
  ...headers,
}
```